src/generator/AutoRest.Go/Extensions.cs

```diff
@@ -20,6 +20,10 @@ public static class Extensions
 
     public const string ReadOnlyConstraint = "ReadOnly";
 
+    private static readonly Regex IsApiVersionPattern = new Regex(@"^api[^a-zA-Z0-9_]?version", RegexOptions.IgnoreCase);
+
+    private static readonly Regex UnwrapAnchorTagsPattern = new Regex("([^<>]*)<a\\s*.*\\shref\\s*=\\s*[\'\"]([^\'\"]*)[\'\"][^>]*>(.*)</a>");
+
```

Comment:
Correct. I agree with the sentiment of keeping things scoped to the method they are used in. This time, though, there is enough value in only spinning up the Regex only once that I think it is worth it. Ideally, we'd even look at adding in the RegexOptions.Compiled flag. I wanted to add a way to evaluate the perf impact before I did that though, because the additional start-up time may not be worth it.